OpenRA.Mods.Common/AI/HarvesterManager.cs

```diff
@@ -0,0 +1,98 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2018 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Mods.Common.Activities;
+using OpenRA.Mods.Common.Pathfinder;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Support;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.AI
+{
+	class HarvesterManager
+	{
+		readonly HackyAI ai;
+		readonly World world;
+		readonly IPathFinder pathfinder;
+		readonly DomainIndex domainIndex;
+		readonly ResourceLayer resLayer;
+		readonly ResourceClaimLayer claimLayer;
+
+		public HarvesterManager(HackyAI ai, Player p)
+		{
+			this.ai = ai;
+			world = p.World;
+			pathfinder = world.WorldActor.Trait<IPathFinder>();
+			domainIndex = world.WorldActor.Trait<DomainIndex>();
+			resLayer = world.WorldActor.TraitOrDefault<ResourceLayer>();
+			claimLayer = world.WorldActor.TraitOrDefault<ResourceClaimLayer>();
+		}
+
+		CPos FindNextResource(Actor actor, Harvester harv)
```

Comment:
Devil's advocate - Is there a reason FindNextResource method doesn't belong as a member of Harvester? (param overhead might be)

I ask, as conceptually, it seems like this is operating in terms of next resource from pov of that harvester.

Reply:
I didn't write this part back when it was introduced, but iirc it's
> containment of AI responsability